packages/vhd-lib/createStreamNbd.js

```diff
@@ -19,13 +21,43 @@ exports.createNbdRawStream = function createRawStream(nbdClient) {
   return stream
 }
 
-exports.createNbdVhdStream = async function createVhdStream(nbdClient, sourceStream, { changedBlocks, vdiInfos } = {}) {
+exports.createNbdVhdStream = async function createVhdStream(
+  nbdClient,
+  sourceStream,
+  {
+    changedBlocks,
+    vdiInfos,
+
+    maxDurationBetweenProgressEmit = MAX_DURATION_BETWEEN_PROGRESS_EMIT,
+    minTresholdPercentBetweenProgressEmit = MIN_TRESHOLD_PERCENT_BETWEEN_PROGRESS_EMIT,
+    onProgress,
+  } = {}
+) {
   const vhd = new VhdNbd(nbdClient, {
     vhdStream: sourceStream,
     vdiInfos,
     changedBlocks,
   })
   await vhd.readHeaderAndFooter()
   await vhd.readBlockAllocationTable()
-  return vhd.stream()
+  let currentBlockRead = 0
+  let lastTimeProgress = 0
+  let lastBlockProgress = 0
+  // add some condition to trigger onProgress only when significant
```

Comment:
Why not put this directly in `VhdAbstract` or as an optional wrapper for `onProgress`?

The logic is not related to NBD.

Reply:
I am not sure, the condition are really related to the transfer, I looked at _.debounce/throttle, but it was not adapted